src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/SplitFileService.cs

```diff
@@ -234,14 +215,13 @@ private static void CopyColumns(DataTable source, DataTable target)
         private static void CopyRows(DataTable target,
             IEnumerable<DataRow> rows,
             List<string> colValues,
-            HashSet<GeographicLevel> importGeographicLevels)
+            DataImport dataImport)
```

Comment:
I know this change gives you access to `.IsSoleGeographicLevel()`, but then you're passing the whole `dataImport` object unnecessarily. 

My general rule of thumb up to now has to try and only pass what is required in arguments, but I've seen you and others make this kind of change in other places. Maybe I'm overestimating the cost of passing a large object between functions / or I am missing something?

Reply:
If you see the latest changes it should make more sense now where I've added it as a property of `DataImport`.

In relation to what you said, if we were passing the argument around between methods across different projects/libraries and that meant the receiving method being called had to be dependent on the same model, which in turn could pull in other unnecessary dependencies, then I can see the benefit of keeping it as separate arguments restricted to basic built-in data types.

In two other places (both in `ImporterService` rather than `SplitFileService`), the `DataImport` is being used as a parameter for 2 or 3 different fields that belong to it, one being this check on the geographic levels.

Rather than there being duplicated code in two places of `ImporterService` and here in `SplitFileService` to work out if the import is for a single level, the DataImport can provide a helper method `HasSoleGeographicLevel`.

The alternative would to get the set of levels from the `DataImport` and count if there's one, in 3 different place.